pdata/pcommon/generated_byteslice.go

```diff
@@ -55,6 +57,17 @@ func (ms ByteSlice) At(i int) byte {
 	return (*ms.getOrig())[i]
 }
 
+// All returns an iterator over index-value pairs in the slice.
+func (ms ByteSlice) All() iter.Seq2[int, byte] {
```

Comment:
Is "All" a canonical name for this? What do other places use?

Reply:
Some discussion about this [here](https://github.com/open-telemetry/opentelemetry-collector/pull/12380#discussion_r1954380804)